### PR TITLE
Topic remove lib splash

### LINKED ISSUE #3357: Removing of libsplash dependency
I will remove the dependency `libsplash` latest on 31.10.2020.

The following plugins should be updated before:
- [x] PhaseSpace - will be updated by @franzpoeschel 
- [x] ParticleCalorimeter @franzpoeschel 
- [x] Radiation - should be updated by @PrometheusPi 
- [x] HDF5Writer - will be removed, please use new plugin `openPMDWriter`
- [x] density profile see: #3515 @sbastrakov 
- [x] plugin MacroParticleCounter @franzpoeschel 

@PrometheusPi @HighIander  Who can do this, who knows the plugin best?

### PR DESCRIPTION
**Pull request by psychocoderHPC**
fix #3357

Remove libSplash dependency and all tools based on libSPlash.

- [x] rebase against #3566
- [x] remove label  `Ci:no-compile`

### CODE REVIEW

File: include/picongpu/plugins/radiation/Radiation.hpp [hunk 1]
```diff
@@ -874,7 +874,7 @@ namespace picongpu
                 void readOpenPMDfile(std::vector<Amplitude>& values, std::string name, const int timeStep)
                 {
                     std::ostringstream filename;
-                    /* add to standard ending added by libSplash for SerialDataCollector */
```
Comment (PrometheusPi) [inline include/picongpu/plugins/radiation/Radiation.hpp:877]: Sorry I missed that.

File: src/tools/bin/pic2xdmf.py [hunk 2]
```diff
@@ -1,608 +0,0 @@
-#!/usr/bin/env python
```
Comment (PrometheusPi) [inline src/tools/bin/pic2xdmf.py:1]: Why did you remocve this program - shouldn't it still work fine with openPMD-api hdf5 output?

File: src/tools/bin/plot_chargeConservation_overTime.py [hunk 3]
```diff
@@ -83,12 +83,7 @@ def deviation_charge_conservation(series, iteration):
             norm = np.max([norm, np.amax(np.abs(species_Density))])
             # add charge density to total charge density
             charge += species_Density
-            # We check the attribute _size of any/all density_[species].
-            # libSplash always keeps this as an array of length 3. It
```
Comment (PrometheusPi) [inline src/tools/bin/plot_chargeConservation_overTime.py:87]: This comment should stay except that: `libSplash` should be replaced by `openPMD-api HDF5`

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
Great pull request. Just a few questions remaining.

File: docs/source/install/profile.rst [hunk 4]
```diff
@@ -69,8 +69,6 @@ Summit (ORNL)
 **Production directory:** usually ``$PROJWORK/$proj/`` (`link <https://www.olcf.ornl.gov/for-users/system-user-guides/summit/summit-user-guide/#file-systems>`_).
 Note that ``$HOME`` is mounted on compute nodes as read-only.
 
-For this profile to work, you need to download the :ref:`PIConGPU source code <install-dependencies-picongpu>` and install :ref:`libSplash and PNGwriter <install-dependencies>` manually.
```
[the remark this answers is not in the document]
Comment (psychocoderHPC) [reply docs/source/install/profile.rst:72]: Ohh my fault, will fix it.

File: src/tools/bin/pic2xdmf.py (hunk 2 above)
Comment (psychocoderHPC) [reply src/tools/bin/pic2xdmf.py:1]: No this tool depends on splash data.

**Comment by psychocoderHPC** [comment]
@PrometheusPi I added your suggestions but keep `pic2xdmf.py` deleted.

File: src/tools/bin/pic2xdmf.py (hunk 2 above)
Comment (PrometheusPi) [reply src/tools/bin/pic2xdmf.py:1]: Okay

**Comment by ax3l** [comment]
Just in case it is needed for a reader, we have to-ASCII export documented for openPMD here (particles): https://openpmd-api.readthedocs.io/en/0.16.1/analysis/pandas.html#openpmd-to-ascii

For fields, it is as easy as reading the [field to numpy](https://openpmd-api.readthedocs.io/en/0.16.1/usage/firstread.html#data) and calling then [.savetxt](https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html).

That said, please don't go via ASCII :sweat_smile: and stay in memory via the Python/C++ APIs :D

**Comment by bussmann** [comment]
> Just in case it is needed for a reader, we have to-ASCII export documented for openPMD here (particles): https://openpmd-api.readthedocs.io/en/0.16.1/analysis/pandas.html#openpmd-to-ascii
> 
> For fields, it is as easy as reading the [field to numpy](https://openpmd-api.readthedocs.io/en/0.16.1/usage/firstread.html#data) and calling then [.savetxt](https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html).
> 
> That said, please don't go via ASCII 😅 and stay in memory via the Python/C++ APIs :D

That warning came too late. Sending you my 10 PByte Ascii output via E-mail right now.

**Pull request merged**